src/BinaryParsers/PEBinary/ProgramDatabase/Pdb.cs

```diff
@@ -348,7 +393,28 @@ public bool ContainsExecutableSectionContribs()
         /// <summary>
         /// Returns the location of the PDB for this module
         /// </summary>
-        public string PdbLocation => this.session.globalScope.symbolsFileName;
+        public string PdbLocation
+        {
+            get
+            {
+                string path = this.session.globalScope.symbolsFileName;
+                if (File.Exists(path))

```

Comment:
Make this if (File.Exists(path) || !Directory.Exists(path)) { return path;}

and you will simplify the remaining code.

Reply:
thanks for the suggestion!

---
In reply to: [563214413](https://github.com/microsoft/binskim/pull/335#discussion_r563214413) [](ancestors = 563214413)